### PR TITLE
feat: support filter by market id in party ordersConnection

### LINKED ISSUE #7216: Support filter by market id in party ordersConnection
# Feature Overview

**In order** to allow user to see his orders for selected market
**We will** update party ordesConnection query
**So that** client applications can retrieved data filtered by partyID and marketId

# Acceptance Criteria
- I can get party orders filter by one or more market ids

# Additional Details
Suggested query
```
query Orders(
  $partyId: ID!
  $pagination: Pagination
  $dateRange: DateRange
  $filter: OrderFilter
  $marketId: ID
) {
  party(id: $partyId) {
    id
    ordersConnection(
      pagination: $pagination
      dateRange: $dateRange
      filter: $filter
      marketId: $marketId
    ) {
      edges {
        node {
          id
        }        
      }
    }
  }
}
```

### PR DESCRIPTION
**Pull request by guoguojin**
Closes #7216

### CODE REVIEW

**Review by jeremyletang — DISMISSED** [review]

File: datanode/gateway/graphql/resolvers.go [hunk 1]
```diff
@@ -1338,6 +1338,7 @@ func (r *myPartyResolver) OrdersConnection(ctx context.Context, party *types.Par
 		Pagination: pagination,
 		DateRange:  dateRange,
 		Filter:     filter,
+		MarketId:   marketId,
```
Comment (MadalinaRaicu) [inline datanode/gateway/graphql/resolvers.go:1341]: (non-blocking): how come marketID is not part of the Filter?

**Review by MadalinaRaicu — DISMISSED** [review]

File: datanode/gateway/graphql/resolvers.go (hunk 1 above)
Comment (guoguojin) [reply datanode/gateway/graphql/resolvers.go:1341]: The filter is used in both ordersConnection under markets, and parties. It wouldn't make sense to include it there.

File: datanode/gateway/graphql/schema.graphql [hunk 2]
```diff
@@ -1926,6 +1926,8 @@ type Party {
     pagination: Pagination
     "Filter orders"
     filter: OrderFilter
+    "Market id to limit the results to"
```
[the remark this answers is not in the document]
Comment (guoguojin) [reply datanode/gateway/graphql/schema.graphql:1929]: Yes, it's optional, the GraphQL definition for it doesn't have a ! suffix so it should be apparent:

```
    "Market ID to limit the results to"
    marketId: ID
```

[the remark this answers is not in the document]
Comment (candida-d) [reply datanode/gateway/graphql/schema.graphql]: Oh yes, forgot about the ! convention

**Review by candida-d — APPROVED** [review]

**Review by wwestgarth — APPROVED** [review]

**Pull request merged**